android/KMEA/app/src/main/java/com/keyman/engine/KMManager.java

```diff
@@ -2052,6 +2069,34 @@ public static void applyKeyboardHeight(Context context, int height) {
     }
   }
 
+  /**
+   * Get the size of the area the window would occupy.
+   * API 30+
+   * https://developer.android.com/reference/android/view/WindowManager#getCurrentWindowMetrics()
+   * @param context
+   * @return Point (width, height)
+   */
+  public static Point getWindowSize(Context context) {
+    WindowManager wm = (WindowManager) context.getSystemService(Context.WINDOW_SERVICE);
+    if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.R) {
+      WindowMetrics windowMetrics = wm.getCurrentWindowMetrics();
+      return new Point(
+        windowMetrics.getBounds().width(),
+        windowMetrics.getBounds().height());
+    }
+
+    // Deprecated in API 30
+    Point size = new Point(0, 0);
+    wm.getDefaultDisplay().getSize(size);
+    return size;
```

Comment:
It's good programming practice to make the default path the main path in the code, so this condition should be inverted:

```suggestion
    if (Build.VERSION.SDK_INT < Build.VERSION_CODES.R) {
      // Deprecated in API 30
      Point size = new Point(0, 0);
      wm.getDefaultDisplay().getSize(size);
      return size;
    }
    
    WindowMetrics windowMetrics = wm.getCurrentWindowMetrics();
    return new Point(
      windowMetrics.getBounds().width(),
      windowMetrics.getBounds().height());    
```